concourse/model/traits/publish.py

```diff
@@ -566,19 +570,15 @@ def validate(self):
             raise ModelValidationError(
                 'must not specify empty list of platforms (omit attr instead)'
             )
+
         resource_names = set()
-        if not self.oci_builder() in (OciBuilder.DOCKER, OciBuilder.DOCKER_BUILDX):
-            for image in self.dockerimages():
-                resource_names.add(image.name)
-                if image.extra_push_targets:
-                    raise ModelValidationError(
-                        f'must not specify extra_push_targets if using {self.oci_builder()}'
-                    )
+        for image in self.dockerimages():
+            resource_names.add(image._base_name)
 
         for helmchart in self.helmcharts:
-            if helmchart.name in resource_names:
+            if helmchart.resource_name in resource_names:
                 raise ModelValidationError(
-                    f'duplicate resource: {helmchart.name=}',
+                    f'duplicate resource: {helmchart.resource_name=}',
                 )
             resource_names.add(helmchart.name)
```

Comment:
Also change to `helmchart.resource_name`

Reply:
unless I am not seeing something - is this not exactly already the case? 🤔